File: backend/models/site_chrome.py
"""Type-safe schema for site-chrome (footer) content.

Wraps the dict in `backend.content.site_chrome` so API responses and
frontend clients share a validated contract.
"""

from typing import List

from pydantic import BaseModel


class FooterLink(BaseModel):
    label: str
    href: str


class SiteFooterContent(BaseModel):
    tagline: str
    nav_links: List[FooterLink]
    legal_links: List[FooterLink]
    mini_links: List[FooterLink]
    brand_label: str
    copyright: str
    social_links: List[FooterLink]
    # Public "call us" line for /partners — empty string hides it.
    contact_phone: str = ""
    # Public contact email — empty string hides it.
    contact_email: str = ""
    # Invitation sentence above the footer's contact details.
    contact_line: str = ""

File: frontend/src/api/siteChrome.ts
/**
 * Site-chrome API client — backend-served footer copy.
 * Consumes /api/site/footer (see backend/api/routes/site.py).
 */
import { API_BASE_URL } from "./client";

export interface FooterLink {
  label: string;
  href: string;
}

export interface SiteFooterContent {
  tagline: string;
  nav_links: FooterLink[];
  legal_links: FooterLink[];
  mini_links: FooterLink[];
  social_links: FooterLink[];
  brand_label: string;
  copyright: string;
  /** Public "call us" line for /partners; empty = hidden. */
  contact_phone: string;
}

export async function loadFooterContent(): Promise<SiteFooterContent> {
  // Footer copy is intentionally public; no Bearer token required.
  const res = await fetch(`${API_BASE_URL}/api/site/footer`);
  if (!res.ok) {
    throw new Error(`Footer content load failed: ${res.status}`);
  }
  return res.json() as Promise<SiteFooterContent>;
}
